### PR TITLE
MOCKITO Fix and refactor tests

### PR DESCRIPTION
**Pull request by D00ktora**
Fixing test structure, extracting preparation into methods. 
Refactoring tests to follow the "AAA" standard.
Separating Arrange/Act/Assert with an empty line.
Refactor naming and spacing of variables to be easily readable. 
Add more test cases to cover edge cases and requirements.

### CODE REVIEW

File: src/test/java/project/service/GasStationServiceImplTest.java [hunk 1]
```diff
@@ -18,76 +18,104 @@
 import java.math.RoundingMode;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.NoSuchElementException;
 
-import static org.mockito.Mockito.when;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
```
Comment (cupakob) [inline src/test/java/project/service/GasStationServiceImplTest.java]: avoid imports with `*`

File: src/test/java/project/service/GasStationServiceImplTest.java [hunk 2]
```diff
@@ -18,76 +18,104 @@
...
 
 @ExtendWith(MockitoExtension.class)
 class GasStationServiceImplTest {
-
+    @InjectMocks
```
Comment (cupakob) [inline src/test/java/project/service/GasStationServiceImplTest.java]: why not only `@Mock`?

File: src/test/java/project/service/GasStationServiceImplTest.java [hunk 3]
```diff
@@ -18,76 +18,104 @@
...
-    void setUp() {
-        gasStationService = new GasStationServiceImpl(repository, modelMapper);
-
+    void setup() {
+        this.gasStationService = new GasStationServiceImpl(repository, modelMapper);
     }
+
     @Test
-    void getStationsByName() {
+    void getStationsByNameTest() {
```
Comment (cupakob) [inline src/test/java/project/service/GasStationServiceImplTest.java:42]: I would reccomend to use block comments for all tests, e.g.

```java
// GIVEN
GasStation testGasStation = createTestGasStation();
when(repository.findByName(testGasStation.getName())).thenReturn(List.of(testGasStation));

// WHEN
List<GasStationDTO> stationsByName = gasStationService.getStationsByName("TestName");

// THEN
assertEquals(1, stationsByName.size());
assertEquals(testGasStation.getName(), stationsByName.get(0).getName());
```

File: src/test/java/project/web/GasStationControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+import project.dto.GasPriceInfoDTO;
+import project.dto.GasStationDTO;
+import project.model.GasStation;
+import project.service.GasStationServiceImpl;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (cupakob) [inline src/test/java/project/web/GasStationControllerTest.java]: here again - avoid `*` imports

File: src/test/java/project/service/GasStationServiceImplTest.java (hunk 1 above)
Comment (D00ktora) [reply src/test/java/project/service/GasStationServiceImplTest.java]: Imports with * removed and added only imports that are used.

File: src/test/java/project/service/GasStationServiceImplTest.java (hunk 2 above)
Comment (D00ktora) [reply src/test/java/project/service/GasStationServiceImplTest.java]: In this case is wrong because i have setup method, when i checkout the file i forgot to correct it. In this class in useless. But in the GasStationControllerTest is useful because it will automatically put inject the mocked class into test class constructor. No need to initialise it manual with setup method.
Its corrected.

File: src/test/java/project/service/GasStationServiceImplTest.java (hunk 3 above)
Comment (D00ktora) [reply src/test/java/project/service/GasStationServiceImplTest.java:42]: Fixed :)

File: src/test/java/project/web/GasStationControllerTest.java (hunk 4 above)
Comment (D00ktora) [reply src/test/java/project/web/GasStationControllerTest.java]: Fixed

**Review by burberius — APPROVED** [review]

File: src/test/java/project/service/GasStationServiceImplTest.java [hunk 5]
```diff
@@ -18,76 +18,128 @@
...
-        Assertions.assertEquals(testGasStation.getName(), stationsByName.get(0).getName());
+
+        // THEN
+        assertEquals(1, stationsByName.size());
+        assertEquals(testGasStation.getName(), stationsByName.get(0).getName());
+    }
+
+    @Test
+    void getStationsByNameTestWithInvalidName() {
+        // GIVEN WHEN
```
Comment (cupakob) [inline src/test/java/project/service/GasStationServiceImplTest.java]: here you can just keep the `// WHEN`

Comment (D00ktora) [reply src/test/java/project/service/GasStationServiceImplTest.java]: GIVEN and WHEN are split to GIVEN and WHEN THEN

**Review by burberius — APPROVED** [review]

**Review by cupakob — APPROVED** [review]
